scripts/e2e-tests.sh

```diff
@@ -246,15 +245,21 @@ function create_test_cluster_with_retries() {
       E2E_CLUSTER_ZONE=${e2e_cluster_zone}
       [[ "${E2E_CLUSTER_ZONE}" == "${zone_not_provided}" ]] && E2E_CLUSTER_ZONE=""
 
-      local geoflag="--gcp-region=${E2E_CLUSTER_REGION}"
-      [[ -n "${E2E_CLUSTER_ZONE}" ]] && geoflag="--gcp-zone=${E2E_CLUSTER_REGION}-${E2E_CLUSTER_ZONE}"
+      local cluster_creation_zone="${E2E_CLUSTER_REGION}"
+      [[ -n "${E2E_CLUSTER_ZONE}" ]] && cluster_creation_zone="${E2E_CLUSTER_REGION}-${E2E_CLUSTER_ZONE}"
+      # Convert cluster version to a full GKE version if we're not using latest
+      if [[ "${E2E_CLUSTER_VERSION}" =~ ^[0-9](\.[0-9]+)?$ ]]; then
+        E2E_CLUSTER_VERSION="v$(gcloud container get-server-config --zone=${cluster_creation_zone} --format="value(validMasterVersions)" 2> /dev/null \
+          | tr ';' '\n' | grep -E ^${E2E_CLUSTER_VERSION} | cut -f1 -d- | sort | tail -1)"
```

Comment:
You don't need a regex here. But to be sure, -F is fine.

```suggestion
          | tr ';' '\n' | grep -F ${E2E_CLUSTER_VERSION} | cut -f1 -d- | sort | tail -1)"
```

Reply:
The reason I used a regex here is to enforce matching the prefix, so that it doesn't happen that `1.19` matching `1.21.19`

Reply:
Touché. :D